### PR TITLE
Update to Remoting 3.15 and Cleanup issues in Channel#current() usages

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change adds some missing null/closing channel checks.
In some cases the change prevents spawning threads if the channel is in the invalid state.

Full diff: https://github.com/jenkinsci/remoting/compare/remoting-3.14...6c4624e3a40d437eed325a2d824c7b92fe5d2b81

Changelog: https://github.com/jenkinsci/remoting/pull/245

No autotests, all changes are edge cases

### Proposed changelog entries

* Major RFE: Update Remoting from 3.14 to 3.15
  * Full changelog: https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md#315
  * Issues to link: JENKINS-48133, JENKINS-48055, JENKINS-37566, JENKINS-48309, JENKINS-47965, JENKINS-48130, JENKINS-37670, JENKINS-46724, JENKINS-38696
* Bug: JENKINS-48309 - Prevent timeout in `FutureImpl#get(timeout)` when a spurious thread wakeup happens before the timeout expiration.
  * Javadoc: http://javadoc.jenkins.io/hudson/model/queue/FutureImpl.html
  * Comment: I think it deserves a separate changelog entry due to the potential impact

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees ping

**Review by ghost — APPROVED** [review]
🐝

File: core/src/main/java/hudson/Launcher.java
Comment (daniel-beck) [inline core/src/main/java/hudson/Launcher.java]: Given the comment above, why is this change needed at all?

Why the variable? Just to better debug?

**Comment by jglick** [comment]
Fix `SetupWizardTest.shouldDisableUnencryptedProtocolsByDefault`…

**Comment by jglick** [comment]
Why is this “on hold”?

**Review by jglick — APPROVED** [review]
Sounds fine.

**Comment by oleg-nenashev** [comment]
> Why is this “on hold”?

Remoting 3.15 still needs to be released

File: core/src/main/java/hudson/Launcher.java
Comment (oleg-nenashev) [reply core/src/main/java/hudson/Launcher.java]: It is explanation why I do not use `getOpenChannelOrFail()`, will update the comment

**Review by ghost — APPROVED** [review]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done. I am about merging it, because I want the changes to be soak tested before we go forward with [JEP-200](https://github.com/jenkinsci/jep/tree/master/jep/200)

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers are you fine with the upgrade assuming that I take responsibility to handle the fallout next week if it happens? Got +1 from @daniel-beck in IRC ("if the changes are safe enough"), maybe somebody else wants to weight in

**Comment by MarkEWaite** [comment]
I am fine with the upgrade.

File: pom.xml
Comment (jglick) [inline pom.xml]: BTW https://github.com/jenkinsci/remoting/pull/245#pullrequestreview-85352695

**Review by jglick — APPROVED** [review]
Fine with me.

File: pom.xml
Comment (oleg-nenashev) [reply pom.xml]: Fixed in https://github.com/jenkinsci/remoting/commit/7f7c94f2a532dd70761dddf4bc027f9fd81766a6

**Pull request merged**
